lucille-plugins/lucille-jlama/src/main/java/com/kmwllc/lucille/jlama/stage/JlamaEmbed.java

```diff
@@ -0,0 +1,174 @@
+import java.util.Iterator;
+import java.util.Optional;
+import org.apache.commons.io.FileUtils;
+import org.apache.zookeeper.common.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This stage uses jlama as an inference machine to download and use the embedding model locally, where it would retrieve the input
+ * to the model from a source field, and set the outputs to a destination field. Note that this stage only supports models that have
+ * safetensor format and BERT architecture.
+ *
+ * Note that in order to use this stage you will need to set this environment variable below, or attach it to your java run command:
+ * export JDK_JAVA_OPTIONS='--add-modules jdk.incubator.vector'
+ *  or
+ * java ... --add-modules jdk.incubator.vector ...
+ *
+ * Config Parameters:
+ * - source (String) : field of which the embedding Stage will retrieve content from
+ * - dest (String, Optional) : name of the field that will store the embeddings, defaults to "embeddings"
+ * - pathToStoreModel (String) : Path where the model will be downloaded to and used
+ * - model (String, Optional) : the name of the embedding model to use in the format of owner/name, retrieved from HuggingFace
+ * - workingMemoryType (String, Optional) : the data type used during runtime for model, impacts accuracy and resource usage.
+ * - workingQuantizationType (String, Optional): the data type used during quantization for model, influences the compression and
+ *   speed of the model, trading off accuracy for performance.
+ * - deleteModelAfter (Boolean, Optional) : whether to delete the model after use, defaults to false
+ *
+ *   DType Options:
+ *    "BOOL" (1 byte)
+ *    "U8" (Unsigned 8-bit integer)
+ *    "I8" (Signed 8-bit integer)
+ *    "I16" (Signed 16-bit integer)
+ *    "U16" (Unsigned 16-bit integer)
+ *    "F16" (16-bit floating point)
+ *    "BF16" (BFloat16, 16-bit floating point)
+ *    "I32" (Signed 32-bit integer)
+ *    "U32" (Unsigned 32-bit integer)
+ *    "F32" (32-bit floating point)
+ *    "F64" (64-bit floating point)
+ *    "I64" (Signed 64-bit integer)
+ *    "U64" (Unsigned 64-bit integer)
+ *    "Q4" (Quantized 4-bit integer)
+ *    "Q5" (Quantized 5-bit integer)
+ */
+public class JlamaEmbed extends Stage {
+  private final String embeddingModel;
+  private final String source;
+  private final String pathToStoreModel;
+  private final String workingMemoryType;
+  private final String workingQuantizationType;
+  private final String dest;
+  private final boolean deleteModelAfter;
+  private AbstractModel model;
+
+  private static final Logger log = LoggerFactory.getLogger(JlamaEmbed.class);
+
+  public JlamaEmbed(Config config) {
+    super(config, new StageSpec()
+        .withRequiredProperties("model", "source", "pathToStoreModel")
+        .withOptionalProperties("workingMemoryType", "workingQuantizationType", "dest", "deleteModelAfter"));
+    pathToStoreModel = config.getString("pathToStoreModel");
```

Comment:
minor: use "this." like the other vars here

Reply:
changed in [a3112e3](https://github.com/kmwtechnology/lucille/pull/201/commits/a3112e36451580e2b2300c7d1cee915f0197e737)